58548_N_K_1_CeVersion_New_Estimate_Board_Cost_for_Combo_Estimate_must…

VarunAgrawalJKT opened the pull request:
…_Calculate_Summarize_for_All_Forms

We have built up a list of like board that will recost the board across all forms on estimate.
New Setting/NK1 - CEAutoRecostBoard have been implemented to automatically recost the board.
Additionally, a "Recost Board" button will be added to the CEPrompt dialog that will allow for this analysis to be done on-demand.

Files:
Source/est/estCostMaterialList.w
Source/est/ RecostBoardEst.p
Source/est/ RecostBoardEst.i

VarunAgrawalJKT committed:
58548_N_K_1_CeVersion_New_Estimate_Board_Cost_for_Combo_Estimate_must_Calculate_Summarize_for_All_Forms

We have built up a list of like board that will recost the board across all forms on estimate.
New Setting/NK1 - CEAutoRecostBoard have been implemented to automatically recost the board.
Additionally, a "Recost Board" button will be added to the CEPrompt dialog that will allow for this analysis to be done on-demand.

Files:
Source/est/estCostMaterialList.w
Source/est/ RecostBoardEst.p
Source/est/ RecostBoardEst.i

brad-advantzware reviewed (commented):
Varun - Don't be discouraged with the amount of feedback I'm giving you here.  This is just to give you an idea of the standards and practices that I would like you to follow.  Overall, you did a good job here.  
One big point of feedback is that I want to avoid copying and pasting code at almost all costs.  We should refactor this procedure and handle the common elements between this and the PO Recost without a pure code copy

VarunAgrawalJKT committed:
Merge branch 'develop' into feature/58548_N_K_1_CeVersion_New_Estimate_Board_Cost_for_Combo_Estimate_must_Calculate_Summarize_for_All_Forms

VarunAgrawalJKT committed:
58548_N_K_1_CeVersion_New_Estimate_Board_Cost_for_Combo_Estimate_must_Calculate_Summarize_for_All_Forms

code refactoring and added the logic to create the New Setting/NK1 - CEAutoRecostBoard in create_nk1.p

Files: est/estCostMaterialList.w
       est/RecostBoardEst.p
       est/RecostBoardEst.i
       sys/inc/create_nk1.p

VarunAgrawalJKT committed:
Merge branch 'develop' into feature/58548_N_K_1_CeVersion_New_Estimate_Board_Cost_for_Combo_Estimate_must_Calculate_Summarize_for_All_Forms

# Conflicts:
#	Source/sys/inc/create_nk1.p

brad-advantzware reviewed (changes requested):
Varun - looks better. Naming conventions for procedures:
PRIVATE - start with lowercase p (eg. pGetNewCosts)
PUBLIC - start with [Program_name]_Procedure (eg. RecostBoardEst_RecostBoard) in the event this procedure is declared SUPER and doesn't require the handle reference.

VarunAgrawalJKT committed:
58548_N_K_1_CeVersion_New_Estimate_Board_Cost_for_Combo_Estimate_must_Calculate_Summarize_for_All_Forms

Code review points have been implemented as per the suggestion, I have renamed the procedure name as per the coding standard.

Files: est/estCostMaterialList.w
       est/RecostBoardEst.p

VarunAgrawalJKT committed:
Merge branch 'develop' into feature/58548_N_K_1_CeVersion_New_Estimate_Board_Cost_for_Combo_Estimate_must_Calculate_Summarize_for_All_Forms

VarunAgrawalJKT committed:
58548_N_K_1_CeVersion_New_Estimate_Board_Cost_for_Combo_Estimate_must_Calculate_Summarize_for_All_Forms

Code Review feedback have been implemented
Files:
Source/est/estCostMaterialList.w
Source/est/RecostBoardEst.i
Source/est/RecostBoardEst.p

VarunAgrawalJKT committed:
58548_N_K_1_CeVersion_New_Estimate_Board_Cost_for_Combo_Estimate_must_Calculate_Summarize_for_All_Forms

Additional Changes

VarunAgrawalJKT committed:
58548_N_K_1_CeVersion_New_Estimate_Board_Cost_for_Combo_Estimate_must_Calculate_Summarize_for_All_Forms

Created recostBoard.png file for Recost Board Button

VarunAgrawalJKT committed:
58548_N_K_1_CeVersion_New_Estimate_Board_Cost_for_Combo_Estimate_must_Calculate_Summarize_for_All_Forms

Additional changes to get the correct score

VarunAgrawalJKT committed:
58548_N_K_1_CeVersion_New_Estimate_Board_Cost_for_Combo_Estimate_must_Calculate_Summarize_for_All_Forms

I have changed the code to get the panel score from eb table.

Files changed: est\RecostBoardEst.p
               system\FormulaProcs.p

VarunAgrawalJKT committed:
58548_N_K_1_CeVersion_New_Estimate_Board_Cost_for_Combo_Estimate_must_Calculate_Summarize_for_All_Forms

Merge the changes from develop branch

VarunAgrawalJKT committed:
Merge branch 'develop' into feature/58548_N_K_1_CeVersion_New_Estimate_Board_Cost_for_Combo_Estimate_must_Calculate_Summarize_for_All_Forms

# Conflicts:
#	Source/est/EstimateCalcProcs.p
#	Source/est/estCostMaterialList.w
#	Source/sys/inc/create_nk1.p

VarunAgrawalJKT committed:
58548_N_K_1_CeVersion_New_Estimate_Board_Cost_for_Combo_Estimate_must_Calculate_Summarize_for_All_Forms

use temp-table instead of database table.

file changed:
est\estCostMaterialList.w
est\EstimateCalcProces.p
est\RecostBoardEst.i
est\RecostBoardEst.p

ron-advantzware committed:
Update RecostBoardEst.i
